### ISSUE TITLE
0.9.0-rc9 cluster follower crash

### ISSUE THREAD

**Issue by grisha** [influxdb/influxdb#1882]
This is http://get.influxdb.org/influxdb-0.9.0_rc9-1.x86_64.rpm on CentOS 6.4.

I have a cluster of 3 nodes, and I've been populating it with a dataset of `df(1)` output going back a couple of years (75K points), 100 at a time. At first everything goes well, but a few minutes into it, both followers crash with the following. I suspect it happens at retention-check-period interval, doesn't seem to be related to the actual data:
```
<influxdb logo asciiart>

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x5520da]

goroutine 88 [running]:
github.com/influxdb/influxdb.(*Server).EnforceRetentionPolicies(0xc208032d00)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/server.go:306 +0x7aa
github.com/influxdb/influxdb.func·010()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/server.go:290 +0xf6
created by github.com/influxdb/influxdb.(*Server).StartRetentionPolicyEnforcement
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/server.go:293 +0x1dd

goroutine 1 [chan receive (nil chan), 10 minutes]:
main.execRun(0xc20800a020, 0x4, 0x4)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:121 +0x780
main.main()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:68 +0x5a3

goroutine 5 [syscall, 10 minutes]:
os/signal.loop()
        /root/.gvm/gos/go1.4/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /root/.gvm/gos/go1.4/src/os/signal/signal_unix.go:27 +0x35

goroutine 6 [chan receive, 10 minutes]:
main.func·002()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:199 +0xed
created by main.startProfiling
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:202 +0x2dc

goroutine 17 [syscall, 10 minutes, locked to thread]:
runtime.goexit()
        /root/.gvm/gos/go1.4/src/runtime/asm_amd64.s:2232 +0x1

goroutine 7 [select]:
github.com/influxdb/influxdb/raft.(*Log).applier(0xc20807a140, 0xc20800a5a0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/log.go:1137 +0x2f0
created by github.com/influxdb/influxdb/raft.(*Log).Open
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/log.go:277 +0x561

goroutine 15 [select]:
github.com/influxdb/influxdb/raft.(*Log).followerLoop(0xc20807a140, 0xc20800a5a0, 0x0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/log.go:686 +0x52c
github.com/influxdb/influxdb/raft.(*Log).stateLoop(0xc20807a140, 0xc20800a5a0, 0x1, 0xc20800a4e0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/log.go:662 +0x39a
created by github.com/influxdb/influxdb/raft.(*Log).startStateLoop
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/log.go:632 +0x98


goroutine 65639 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 90 [IO wait]:
net.(*pollDesc).Wait(0xc20800ec30, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800ec30, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20800ebd0, 0xc20806e000, 0x1000, 0x1000, 0x0, 0x7fa6b12d1ac8, 0xc2083cb1d8)
        /root/.gvm/gos/go1.4/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208036488, 0xc20806e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7fa6b12d3490, 0xc208036488, 0xc208062478, 0xc20806e000, 0x1000, 0x1000, 0x46, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc20801d300, 0xc20806e000, 0x1000, 0x1000, 0x46, 0x0, 0x0)
        <autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc2080dab40)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc2080dab40, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:295 +0x257
net/http/internal.readLine(0xc2080dab40, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/internal/chunked.go:110 +0x5a
net/http/internal.(*chunkedReader).beginChunk(0xc2080a9920)
        /root/.gvm/gos/go1.4/src/net/http/internal/chunked.go:47 +0x46
net/http/internal.(*chunkedReader).Read(0xc2080a9920, 0xc208351de0, 0x16, 0x16, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/internal/chunked.go:77 +0xbb
net/http.(*body).readLocked(0xc2080f0580, 0xc208351de0, 0x16, 0x16, 0xffffffff, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transfer.go:584 +0x7a
net/http.(*body).Read(0xc2080f0580, 0xc208351de0, 0x16, 0x16, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transfer.go:579 +0x115
net/http.(*bodyEOFSignal).Read(0xc2080f05c0, 0xc208351de0, 0x16, 0x16, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:1193 +0x285
io.ReadAtLeast(0x7fa6b12d3690, 0xc2080f05c0, 0xc208351de0, 0x16, 0x16, 0x16, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/io/io.go:298 +0xf1
io.ReadFull(0x7fa6b12d3690, 0xc2080f05c0, 0xc208351de0, 0x16, 0x16, 0x2a, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/io/io.go:316 +0x6d
github.com/influxdb/influxdb/messaging.(*MessageDecoder).Decode(0xc20809d750, 0xc2083f8000, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/messaging/broker.go:1156 +0xa4
github.com/influxdb/influxdb/messaging.func·012()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/messaging/client.go:424 +0x113
created by github.com/influxdb/influxdb/messaging.(*Client).streamFromURL
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/messaging/client.go:434 +0x64a
goroutine 16 [IO wait]:
net.(*pollDesc).Wait(0xc20800f560, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800f560, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20800f500, 0xc2080ca000, 0x1000, 0x1000, 0x0, 0x7fa6b12d1ac8, 0xc2083cb480)
        /root/.gvm/gos/go1.4/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208036280, 0xc2080ca000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7fa6b12d3490, 0xc208036280, 0xc208062738, 0xc2080ca000, 0x1000, 0x1000, 0x5e, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc20801d660, 0xc2080ca000, 0x1000, 0x1000, 0x5e, 0x0, 0x0)
        <autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc20800b920)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc20800b920, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:295 +0x257
net/http/internal.readLine(0xc20800b920, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/internal/chunked.go:110 +0x5a
net/http/internal.(*chunkedReader).beginChunk(0xc2080a9710)
        /root/.gvm/gos/go1.4/src/net/http/internal/chunked.go:47 +0x46
net/http/internal.(*chunkedReader).Read(0xc2080a9710, 0xc20846e1c0, 0x1, 0x18, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/internal/chunked.go:77 +0xbb
net/http.(*body).readLocked(0xc208035440, 0xc20846e1c0, 0x1, 0x18, 0xffffffff, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transfer.go:584 +0x7a
net/http.(*body).Read(0xc208035440, 0xc20846e1c0, 0x1, 0x18, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transfer.go:579 +0x115
net/http.(*bodyEOFSignal).Read(0xc208035480, 0xc20846e1c0, 0x1, 0x18, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:1193 +0x285
io.ReadAtLeast(0x7fa6b12d3690, 0xc208035480, 0xc20846e1c0, 0x1, 0x18, 0x1, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/io/io.go:298 +0xf1
io.ReadFull(0x7fa6b12d3690, 0xc208035480, 0xc20846e1c0, 0x1, 0x18, 0xc2083f8930, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/io/io.go:316 +0x6d
github.com/influxdb/influxdb/raft.(*LogEntryDecoder).Decode(0xc20809d0f0, 0xc2083f8930, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/encoder.go:48 +0xb8
github.com/influxdb/influxdb/raft.(*Log).ReadFrom(0xc20807a140, 0x7fa6b12d3660, 0xc208035480, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/log.go:1595 +0x1dc
github.com/influxdb/influxdb/raft.(*Log).readFromLeader(0xc20807a140, 0xc20801c9c0, 0xc20800a900)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/log.go:744 +0x59d
created by github.com/influxdb/influxdb/raft.(*Log).followerLoop
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/log.go:683 +0x18e

goroutine 65778 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 19 [IO wait, 10 minutes]:
net.(*pollDesc).Wait(0xc20800edf0, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800edf0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20800ed90, 0x0, 0x7fa6b12d1ac8, 0xc20809ce00)
        /root/.gvm/gos/go1.4/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2080360f0, 0xc20802ee08, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc2080360f0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc20800aae0, 0x7fa6b12d39f0, 0xc2080360f0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7fa6b12d39f0, 0xc2080360f0, 0x7fa6b12d3a28, 0xc20801ca40, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/server.go:1606 +0xa1
main.func·003()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/cmd/influxd/run.go:56 +0x77
created by main.Run
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/cmd/influxd/run.go:56 +0x6cd

goroutine 20 [select]:
github.com/influxdb/influxdb.(*Broker).continuousQueryLoop(0xc20803bbf0, 0xc20800aa20)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/broker.go:75 +0x10a
created by github.com/influxdb/influxdb.(*Broker).RunContinuousQueryLoop
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/broker.go:55 +0x78

goroutine 30 [IO wait]:
net.(*pollDesc).Wait(0xc20800f250, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800f250, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20800f1f0, 0xc2080b6000, 0x1000, 0x1000, 0x0, 0x7fa6b12d1ac8, 0xc2083cab00)
        /root/.gvm/gos/go1.4/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2080361e8, 0xc2080b6000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/net.go:121 +0xdc
net/http.(*liveSwitchReader).Read(0xc208038908, 0xc2080b6000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/server.go:214 +0xab
io.(*LimitedReader).Read(0xc20801d440, 0xc2080b6000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/io/io.go:408 +0xce
bufio.(*Reader).fill(0xc20800b380)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc20800b380, 0xc20802fb0a, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:295 +0x257
bufio.(*Reader).ReadLine(0xc20800b380, 0x0, 0x0, 0x0, 0xffffffffffffff00, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:324 +0x62
net/textproto.(*Reader).readLineSlice(0xc20832e030, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/textproto/reader.go:55 +0x9e
net/textproto.(*Reader).ReadLine(0xc20832e030, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/textproto/reader.go:36 +0x4f
net/http.ReadRequest(0xc20800b380, 0xc2083c8a90, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/request.go:598 +0xcb
net/http.(*conn).readRequest(0xc2080388c0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/server.go:586 +0x26f
net/http.(*conn).serve(0xc2080388c0)
        /root/.gvm/gos/go1.4/src/net/http/server.go:1162 +0x69e
created by net/http.(*Server).Serve
        /root/.gvm/gos/go1.4/src/net/http/server.go:1751 +0x35e

goroutine 65702 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65833 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 35 [select, 10 minutes]:
net/http.(*persistConn).readLoop(0xc2080626e0)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
        /root/.gvm/gos/go1.4/src/net/http/transport.go:660 +0xc9f

goroutine 36 [select, 10 minutes]:
net/http.(*persistConn).writeLoop(0xc2080626e0)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
        /root/.gvm/gos/go1.4/src/net/http/transport.go:661 +0xcbc

goroutine 65810 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65756 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65712 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 86 [select, 10 minutes]:
github.com/influxdb/influxdb/messaging.(*Client).streamFromURL(0xc20804c7d0, 0xc20800e690, 0xc20800a300, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/messaging/client.go:437 +0x829
github.com/influxdb/influxdb/messaging.(*Client).streamer(0xc20804c7d0, 0xc20800a300)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/messaging/client.go:389 +0x150
created by github.com/influxdb/influxdb/messaging.(*Client).Open
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/messaging/client.go:143 +0x27e

goroutine 83 [select, 10 minutes]:
net/http.(*persistConn).readLoop(0xc208062420)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
        /root/.gvm/gos/go1.4/src/net/http/transport.go:660 +0xc9f

goroutine 84 [select, 10 minutes]:
net/http.(*persistConn).writeLoop(0xc208062420)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
        /root/.gvm/gos/go1.4/src/net/http/transport.go:661 +0xcbc

goroutine 87 [select]:
github.com/influxdb/influxdb.(*Server).processor(0xc208032d00, 0x7fa6b12d4040, 0xc20804c7d0, 0xc20800a3c0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/server.go:2672 +0x2aa
created by github.com/influxdb/influxdb.(*Server).setClient
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/server.go:351 +0x132

goroutine 89 [IO wait, 10 minutes]:
net.(*pollDesc).Wait(0xc20800ebc0, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800ebc0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20800eb60, 0x0, 0x7fa6b12d1ac8, 0xc20809d538)
        /root/.gvm/gos/go1.4/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2080361d8, 0xd7d0a0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc2080361d8, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc20800aba0, 0x7fa6b12d39f0, 0xc2080361d8, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7fa6b12d39f0, 0xc2080361d8, 0x7fa6b12d41f8, 0xc20809d640, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/admin.(*Server).ListenAndServe(0xc2080a83c0)
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/admin/admin.go:38 +0x1d8
created by main.Run
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/cmd/influxd/run.go:103 +0x118e

goroutine 65832 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65734 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 1015 [IO wait]:
net.(*pollDesc).Wait(0xc2080fc3e0, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2080fc3e0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc2080fc380, 0xc2080e7000, 0x1000, 0x1000, 0x0, 0x7fa6b12d1ac8, 0xc2083ca8b8)
        /root/.gvm/gos/go1.4/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208036918, 0xc2080e7000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7fa6b12d3490, 0xc208036918, 0xc208062d68, 0xc2080e7000, 0x1000, 0x1000, 0xc208498600, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc20810ef00, 0xc2080e7000, 0x1000, 0x1000, 0xc208010000, 0x0, 0x0)
        <autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc2080da6c0)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc2080da6c0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc208062d10)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
        /root/.gvm/gos/go1.4/src/net/http/transport.go:660 +0xc9f

goroutine 65767 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65799 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65745 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65788 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65821 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65691 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 65682 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 1016 [select]:
net/http.(*persistConn).writeLoop(0xc208062d10)
        /root/.gvm/gos/go1.4/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
        /root/.gvm/gos/go1.4/src/net/http/transport.go:661 +0xcbc

goroutine 65723 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
        /root/.gvm/pkgsets/go1.4/global/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb
```
The config file is:

```

bind-address = "0.0.0.0"

reporting-disabled = true

[initialization]
# Comma-delimited URLs, in the form http://host:port, for joining another cluster.
join-urls = "" ### <- these vary from node to node

[authentication]
enabled = false

[admin]
enabled = true
port = 8083

[api]
# ssl-port = 8087    # SSL support is enabled if you set a port and cert
# ssl-cert = "/path/to/cert.pem"

[[graphite]]
enabled = false

[collectd]
enabled = false

[udp]
enabled = false

[broker]
dir  = "/var/lib/influxdb/raft"
port = 8086

[data]
dir = "/var/lib/influxdb/db"
port = 8086
retention-check-enabled = true
retention-check-period = "10m"

[cluster]
dir = "/var/influxdb/state"

[logging]
file   = "/var/log/influxdb/influxd.log"
write-tracing = false
```
The database was created with:

```
curl -G http://echo04:8604/query --data-urlencode "q=CREATE DATABASE foo"
curl -G http://echo04:8604/query --data-urlencode "q=CREATE RETENTION POLICY bar ON foo DURATION 3d REPLICATION 1 DEFAULT"
```
Sample data (written via a ruby script using `Net::HTTP`):

```
{"database"=>"foo", "retentionPolicy"=>"bar", 
"points"=>[{"name"=>"disk", "tags"=>{"cluster"=>"baz"}, "timestamp"=>"2012-11-20T00:15:01.000Z",
   "fields"=>{"total"=>106327242178560, 
                   "used"=>34211910079546, 
                   "free"=>66669198299136, 
                   "percent"=>32}}, 
 <...repeat 99 times....> 
]}
```

**Comment by otoolep** [comment]
Thanks for the excellent bug report @grisha -- I believe the issue should be addressed by PR #1883.

**Comment by otoolep** [comment]
I am hopeful this issue is fixed. Version RC10 will have the change that should address this, so please let us know if you still hit this issue.
